### PR TITLE
Add hot-reload go server and streamline Makefile

### PR DESCRIPTION
**Pull request by macrael**
I've tweaked our development environment a bit, take a look at the diff to the README
for a sense of the consequences. It should mean that we have to keep track of 
fewer targets in the normal course of things.

I've also added a replacement for the go flags package that reads env vars in addition to command line flags to make our app more 12 factor friendly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,7 +44,16 @@ All of our code is intermingled in the top level directory of mymove. Here is an
 `src`: The react source code for the client \
 `vendor`: Cached dependencies for the server
 
-### Prerequisites
+### Setup: Initial Setup
+
+The following commands will get mymove running on your machine for the frist time. Please read below for explainations of each of the commands.
```
Comment (tinyels) [inline README.md]: explainations => explanations

**Comment by macrael** [comment]
#100 😎

File: README.md [hunk 2]
```diff
@@ -55,32 +64,31 @@ All of our code is intermingled in the top level directory of mymove. Here is an
...
 
 You can verify the server is working as follows:
 
-`> curl http://localhost:8080/api/v1/issues --data "{ \"body\": \"This is a test issue\"}"`
+`> curl http://localhost:8080/api/v1/issues --data "{ \"description\": \"This is a test issue\"}"`
 
 from which the response should be like
 
-`{"id":"d5735bc0-7553-4d80-a42d-ea1e50bbcfc4", "body": "This is a test issue", "created_at": "2018-01-04 14:47:28.894988", "updated_at": "2018-01-04 14:47:28.894988"}`
+`{"id":"d5735bc0-7553-4d80-a42d-ea1e50bbcfc4", "description": "This is a test issue", "created_at": "2018-01-04 14:47:28.894988", "updated_at": "2018-01-04 14:47:28.894988"}`
```
Comment (ntwyman) [inline README.md]: In future - can you separate out disjoint PR's like this. This is just a bug fix to the documentation and not part of this main PR

File: cmd/webserver/main.go [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
 import (
-	"flag"
+	"fmt"
 	"log"
 	"net/http"
 	"path"
 
 	"github.com/go-openapi/loads"
 	"github.com/markbates/pop"
+	"github.com/namsral/flag"
```
Comment (ntwyman) [inline cmd/webserver/main.go]: A comment about why we're using this flag drop in would be useful. I had to go lookup the github repo to find out.

File: README.md [hunk 4]
```diff
@@ -55,32 +64,31 @@ All of our code is intermingled in the top level directory of mymove. Here is an
...
 
 ### Setup: Server
 
-1. `make server_run`: installs dependencies and builds both the client and the server, then runs the server.
+1. `make server_run`: installs dependencies, then builds and runs the server using `gin`, which is a hot reloading go server. It will listen on 8080 and will rebuild the actual server any time a go file changes. Pair this with `make client_run` to have hot reloading of the entire application.
 
-For faster development, use `make server_run_dev`. This builds and runs the server but skips updating dependences and re-building the client. Those tasks can be accomplished as needed with `make server_deps` and `make client_build`
+In rare cases, you may want to run the server standalone, in which case you can run `make server_run_standalone`. This will build both the client and the server and this invocation can be relied upon to be serving the client js on its own rather than relying on webpack doing so as when you run `make client_run`. You can run this without running `mke client_run` and the whole app should work.
```
Comment (ntwyman) [inline README.md]: s/mke/make

File: README.md (hunk 2 above)
Comment (macrael) [reply README.md]: 👍

**Pull request merged**